includes/admin/class-wc-admin-taxonomies.php

```diff
@@ -49,6 +51,7 @@ public function __construct() {
 
 		// Category/term ordering.
 		add_action( 'create_term', array( $this, 'create_term' ), 5, 3 );
+		add_action( 'delete_product_cat', array( $this, 'assign_default_product_cat' ) );
```

Comment:
Adding a new public function for the sole purpose of invoking a method on a private class kind of defeats the whole purpose of making that class private... what about removing the function and adding a direct callback instead? Something like this:

```suggestion
		add_action(
			'delete_product_cat',
			function() {
				wc_get_container()->get( AssignDefaultCategory::class )->schedule_action();
			},
		);
```